src/inmanta/agent/io/remote.py

```diff
@@ -125,10 +129,19 @@ def _build_connect_string(self) -> str:
 
     def _execute(self, function_name, *args, **kwargs):
         with self._lock:
-            ch = self._gw.remote_exec(local)
-            ch.send((function_name, args, kwargs))
-            result = ch.receive()
-            ch.close()
+            try:
+                ch = self._gw.remote_exec(local)
+            except OSError:
+                raise ChannelClosedException(
```

Comment:
This call will fail, because the ChannelClosedException exception doesn't support keyword arguments in its constructor.

Reply:
resolved